src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -23,11 +23,127 @@ Copyright (c) .NET Foundation. All rights reserved.
-                            RestoreForComposeCache;
-                            ComputeAndCopyFilesToCacheDirectory"/>
+          DependsOnTargets="_InitializeBasicProps;
+                            PrepareForComposeCache;
+                            PrepOptimizer;
+                            CacheWorkerMain;
+                            CacheFinalizer;"/>
+
+
+  <!--
+    ============================================================
+                                        CacheWorkerMain
+
+   Spins of the compose mappers
+    ============================================================
+    -->
+
+  <Target Name="CacheWorkerMain">
+    
+    <ItemGroup>
+      <_AllProjects Include="$(AdditionalProjects.Split('%3B'))"/>
+      <_AllProjects Include ="$(MSBuildProjectFullPath)"/>
+    </ItemGroup>
+    
+    <MSBuild Projects="%(_AllProjects.Identity)"
+                 Targets="CacheWorkerMapper"
+                 Properties="ComposeWorkingDir=$(ComposeWorkingDir);
+                             PublishDir=$(PublishDir);
+                             TargetFramework=$(_TFM);
+                             _JitPath=$(_JitPath);
+                             _Crossgen=$(_Crossgen);
+                             DisableImplicitFrameworkReferences=true;
+                             SkipUnchangedFiles=$(SkipUnchangedFiles);
+                             PreserveCacheLayout=$(PreserveCacheLayout)"
+                  ContinueOnError="WarnAndContinue"  />
+   </Target>
+  <!--
+    ============================================================
+                                        CacheWorkerMapper
+
+   Spins of the compose mappers
+    ============================================================
+    -->
+
+  <Target Name="CacheWorkerMapper">
+    
+    <MSBuild Projects="$(MSBuildProjectFullPath)"
+                 Targets="CacheWorker"
+                 Properties="_RPackageName=%(PackageReference.Identity);
+                             _RPackageVersion=%(PackageReference.Version);
+                             ComposeWorkingDir=$(ComposeWorkingDir);
+                             PublishDir=$(PublishDir);
+                             TargetFramework=$(TargetFramework);
+                             _JitPath=$(_JitPath);
+                             _Crossgen=$(_Crossgen);
+                             SkipUnchangedFiles=$(SkipUnchangedFiles);
+                             PreserveCacheLayout=$(PreserveCacheLayout)"
+                  ContinueOnError="WarnAndContinue"
+             />
+  </Target>
+
+
+  <Target Name="CacheWorker"
+          DependsOnTargets="_InitializeBasicProps;
+                            PrepforRestoreForComposeCache;
+                            CacheWorkerPerformWork"/>
+
+
+  <Target Name="CacheWorkerPerformWork"
+          DependsOnTargets="RestoreForComposeCache;
+                            ComputeAndCopyFilesToCacheDirectory;"
+          Condition="!Exists($(CacheWorkerWorkingDir))" />
+    <!--
```

Comment:
Nit: more whitespace issues. Avoid more than one blank line. And here the start of the comment for the next target looks like it's nested in the previous target as there is no blank line between them and the indentation is off.